src/coreclr/jit/gentree.h

```diff
@@ -8536,6 +8544,24 @@ struct GenTreeCC final : public GenTree
 #endif // DEBUGGABLE_GENTREE
 };
 
+// Represents a node with two operands and a condition.
+struct GenTreeOpCC final : public GenTreeOp
```

Comment:
Is there any difference between a GenTreeOpCC and a GenTreeConditional?

If not, I assume it's here just to help with categorising nodes.
Could GenTreeOpCC be a subclass of GenTreeConditional (or vice versa?)

Reply:
Yes -- `GenTreeOpCC` is a binary node with a condition code in it. As such, it only makes sense in the backend with explicit ordering where the flags definition can happen right before it. There is no explicit link to the flags definition (at least until #82355, though this is just an experiment for now).
So it's sort of a lowered version of `GenTreeConditional`.

Reply:
I didn't spot this at first but `GenTreeOpCC` does not have a pointer to the genTree for the condition, except via gtPrev. I see that `GenTreeCC` works in the same way. That feels odd at first, but this is LIR, and it's the prev/next that matter. Ok, I'm happy with how that works now.


Thinking about the CCMP nodes, they would also use `GenTreeOpCC`. We'd need a `CCMP` node plus all 6 conditions (`CCMP_EQ`, `CCMP_NE`, `CCMP_LT`, etc).

Coming from HIR we have:
```
JTRUE(NE(AND(LT, GT), 0))
```

OptimizeConstCompare() will lower to a TEST_EQ:
```
JTRUE(TEST_EQ(LT, GT)))
```

Then lowerCompare() will create a CCMP and CMP:
```
CMP
JTRUE(CCMP_LT[GT])
```

Finally LowerJtrue() will create the JCC:
```
CMP
CCMP[GT]
JCC[LT]
```

Which means I need to wait for this PR before I can continue with the CCMP work.

Reply:
That sounds good to me. Feel free to lift any of the code from this PR if it helps.

> We'd need a CCMP node plus all 6 conditions (CCMP_EQ, CCMP_NE, CCMP_LT, etc).

Note that this PR is teaching `TryLowerConditionToFlagsNode` about `GT_SETCC` nodes. I think I had a note on it somewhere else, but I wouldn't expect the 6 combinations to be necessary due to that. I.e. we can represent `CCMP_LT` as `CCMP + SETCC[LT]`. The transformation done in this PR will automatically lower `JTRUE(SETCC[LT])  into `JCC[LT]`.
I think it's likely compare chains can end up being handled similarly to this, that is, something like `NE(SETCC[cond], x)` can be turned into some shape of `CCMP + SETCC` and then lowering will continue working on these successively.

Reply:
With that scheme:

Coming from HIR we have:
```
JTRUE(NE(AND(LT, GT), 0))
```

OptimizeConstCompare() will lower to a TEST_EQ:
```
JTRUE(TEST_EQ(LT, GT)))
```

Then lowerCompare() will create a CCMP, SETCC and CMP:
```
CMP
CCMP[GT]
JTRUE(SETCC[LT])
```

Finally LowerJtrue() will create the JCC:
```
CMP
CCMP[GT]
JCC[LT]
```

I'm happy with that setup.

It's not clear to me how much of the existing AND chains code will be left in codegen after this, but that can be checked later.